types/block_test.go

```diff
@@ -86,10 +86,15 @@ func TestBlockValidateBasic(t *testing.T) {
 		{"Tampered EvidenceHash", func(blk *Block) {
 			blk.EvidenceHash = []byte("something else")
 		}, true},
+		{"ConflictingHeadersEvidence", func(blk *Block) {
+			blk.Evidence = EvidenceData{Evidence: []Evidence{&ConflictingHeadersEvidence{}}}
+		}, true},
+		{"PotentialAmnesiaEvidence", func(blk *Block) {
+			blk.Evidence = EvidenceData{Evidence: []Evidence{&PotentialAmnesiaEvidence{}}}
+		}, true},
 	}
 	for i, tc := range testCases {
 		tc := tc
```

Comment:
Is this line necessary as well

Reply:
yes, otherwise you get Using the variable on range scope `i` in function literal (scopelint)